parsec/core/fs/workspacefs/versioning_helpers.py

```diff
@@ -299,14 +304,14 @@ class ManifestCacheCounter:
         ManifestCacheDownloadLimitReached
     """
 
-    def __init__(self, manifest_cache: ManifestCache, limit: int):
+    def __init__(self, manifest_cache: ManifestCache, limit: Optional[int]):
```

Comment:
would be more elegant to have `limit: int = math.inf` (this way no need to explicitly pass `limit=None`)
However I guess `math.inf` is not friend with `int` type so keep it this way might be justified...

Reply:
As you says, `math.inf` is seen as a float.
```
 Incompatible default for argument "limit" (default has type "float", argument has type "int")
```